contractcourt/briefcase.go

```diff
@@ -1184,6 +1202,137 @@ func (b *boltArbitratorLog) checkpointContract(c ContractResolver,
 	}, func() {})
 }
 
+// InsertCanceledHTLCs inserts the given set of canceled HTLCs into the
+// database. It converts the set to a slice and persists the slice.
+//
+// NOTE: Part of the ArbitratorLog interface.
+func (b *boltArbitratorLog) InsertCanceledHTLCs(
+	canceledHTLCs fn.Set[uint64]) error {
+
+	return kvdb.Batch(b.db, func(tx kvdb.RwTx) error {
+		scopeBucket, err := tx.CreateTopLevelBucket(b.scopeKey[:])
+		if err != nil {
+			return err
+		}
+
+		var w bytes.Buffer
+		err = encodeHtlcSet(&w, canceledHTLCs)
+		if err != nil {
+			return err
+		}
+
+		return scopeBucket.Put(canceledHTLCsKey, w.Bytes())
+	})
+}
+
+// FetchCanceledHTLCs fetches the set of canceled HTLCs from the database.
+//
+// NOTE: Part of the ArbitratorLog interface.
+func (b *boltArbitratorLog) FetchCanceledHTLCs() (fn.Set[uint64], error) {
+	canceledHTLCs := fn.NewSet[uint64]()
+
+	err := kvdb.View(b.db, func(tx kvdb.RTx) error {
+		scopeBucket := tx.ReadBucket(b.scopeKey[:])
+		if scopeBucket == nil {
+			return errScopeBucketNoExist
+		}
+
+		canceledHTLCsBytes := scopeBucket.Get(canceledHTLCsKey)
+		if canceledHTLCsBytes == nil {
+			return errNoCanceledHTLCs
```

Comment:
it is, because after the kvdb.View, we only return an error if:

```
if err != nil && !errors.Is(err, errNoCanceledHTLCs) &&
		!errors.Is(err, errScopeBucketNoExist) {

		return nil, err
	}
```

Reply:
so i guess it's equivalent of returning an nil here?
```go
if canceledHTLCsBytes == nil {
    return nil
}
```